src/bridge/EspressoTEEVerifier.sol

```diff
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {
+    V3QuoteVerifier
+} from "@automata-network/dcap-attestation/contracts/verifiers/V3QuoteVerifier.sol";
+import {BELE} from "@automata-network/dcap-attestation/contracts/utils/BELE.sol";
+import {Header} from "@automata-network/dcap-attestation/contracts/types/CommonStruct.sol";
+import {
+    IQuoteVerifier
+} from "@automata-network/dcap-attestation/contracts/interfaces/IQuoteVerifier.sol";
+import {
+    HEADER_LENGTH,
+    ENCLAVE_REPORT_LENGTH
+} from "@automata-network/dcap-attestation/contracts/types/Constants.sol";
+import {EnclaveReport} from "@automata-network/dcap-attestation/contracts/types/V3Structs.sol";
+/**
+ *
+ * @title  Verifies quotes from the TEE and attests on-chain
+ * @notice Contains the logic to verify a quote from the TEE and attest on-chain. It uses the V3QuoteVerifier contract
+ *         to verify the quote. Along with some additional verification logic.
+ */
+
+contract EspressoTEEVerifier is V3QuoteVerifier {
+    constructor(address _router) V3QuoteVerifier(_router) {}
+
+    /**
+        @notice Verify a quote from the TEE and attest on-chain
+        @param rawQuote The quote from the TEE
+        @return success True if the quote was verified and attested on-chain
+     */
+    function verify(bytes calldata rawQuote) external view returns (bool success) {
+        // Parse the header
+        Header memory header = _parseQuoteHeader(rawQuote);
```

Comment:
may I ask why can't we directly accept a header and push the responsibility of parsing the header outside the contract? isn't this more efficient?

Reply:
I might be wrong but this may have been added because we want to later verify some properties about the header in this contract.

Reply:
The function is private in the automata repo: https://github.com/EspressoSystems/automata-dcap-attestation/blob/6463b4b3cd9f18b795a4e8b672231ca83af3230e/contracts/AutomataDcapAttestation.sol#L84

We could submit a PR to make it public. But I'm kind of curious now why this is private in the first place.

Reply:
I didnt want to change their code because they probably have already audited it.

Reply:
@alxiong   The enclave signs on the data that is attached to the quote, so if we parse the quote and get the header, we know for sure that the header is correct because the enclave signed on it. Even if we let the user send the header, we would still have to parse the header from within the quote and verify it, so it doesnt make sense